### PR TITLE
RCT/YJ/get_HVAC_systems_primarily_serving_comp_rooms

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**

### CODE REVIEW

File: docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md [hunk 1]
```diff
@@ -3,33 +3,32 @@
...
 
 1. get_hvac_zone_list_w_area()  
 
 ## Logic:   
-- Create RMR object from RMR function input (RMR = function input): `X_RMR = RMR`
-- Get dictionary list of zones and the total floor area served by each HVAC system: `hvac_zone_list_w_area_dict_x = get_hvac_zone_list_w_area (X_RMR)`
-- For each hvac_x in the X_RMR: `for hvac_x in X_RMR...HeatingVentilatingAirConditioningSystem:`
-    - Get the total floor area served by the hvac system: `hvac_sys_total_floor_area_x = hvac_zone_list_w_area_dict_x[hvac_x.id]["TOTAL_AREA"]`
+- Get dictionary list of zones and the total floor area served by each HVAC system: `hvac_zone_list_w_area_dict = get_hvac_zone_list_w_area(RMD)`
+- For each hvac_x in the X_RMD: `for hvac in RMD...HeatingVentilatingAirConditioningSystem:`
```
Comment (Jiarongx-Xie) [inline docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md:17]: `for hvac in RMD...HeatingVentilatingAirConditioningSystem:`, change `hvac` to `hvac_x`

File: docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md [hunk 2]
```diff
@@ -3,33 +3,32 @@
...
+    - Get the total floor area served by the hvac system: `hvac_sys_total_floor_area = hvac_zone_list_w_area_dict_x[hvac_x.id]["total_area"]`
     - Reset hvac system includes computer room boolean variable: `hvac_system_serves_computer_room_space = FALSE` 
     - Reset hvac_sys_computer_room_floor_area variable: `hvac_sys_computer_room_floor_area = 0`
-    - Get list of zones served by hvac_x: `hvac_zone_list_x = hvac_zone_list_w_area_dict_x[hvac_x.id]["ZONE_LIST"]`
-    - For each zone_x in hvac_zone_list_x: `for zone_x in hvac_zone_list_x:`
-        - For each space in zone: `for space_x in zone_x.Spaces:`        
-            - Check if space is of type computer room, if yes then set the hvac system has computer room boolean variable to true and add the floor area to the total computer room floor area variable for the hvac system: `if space_x.lighting_space_type in ["COMPUTER_ROOM"]:`
+    - Get list of zones served by hvac_x: `hvac_zone_list = hvac_zone_list_w_area_dict_x[hvac.id]["zone_list"]`
+    - For each zone_x in hvac_zone_list_x: `for zone in hvac_zone_list:`
```
Comment (Jiarongx-Xie) [inline docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md:22]: `For each zone_x in hvac_zone_list_x` same comment here

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+COMPUTER_ROOM_RATIO_THRESHOLD = 0.5
+
+
+def get_HVAC_systems_primarily_serving_comp_rooms(rmd: dict) -> list[str]:
+    """
+    Returns a list of HVAC systems in which greater than 50% of the area served by the HVAC system is computer room space.
+
+    Parameters
+    ----------
+    rmd: json
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.py:20]: should we change this to `rmd: dict` to keep consistency?

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    """
+
+    # get each hvac system's serving area and zone list
+    hvac_zone_list_w_area_dict = get_hvac_zone_list_w_area_by_rmi_dict(rmd)
+
+    hvac_systems_primarily_serving_comp_rooms_list = []
+    for hvac_id in find_all(
+        "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id",
+        rmd,
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.py:36]: Why not use `for hvac_id in hvac_zone_list_w_area_dict`?

File: docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md [hunk 5]
```diff
@@ -3,33 +3,32 @@
...
     - Check if the hvac system serves computer room boolean variable equals TRUE, if true then proceed otherwise loop to next hvac system: `if hvac_system_serves_computer_room_space == TRUE:`
-        - Check if computer room floor area is greater than 50% of the total floor area served by the hvac system, if yes then carry on with the check: `if hvac_sys_computer_room_floor_area/hvac_sys_total_floor_area_x > 0.5:` 
-            - Add to the list of hvac systems, hvac_systems_primarily_serving_comp_rooms_list_x: `hvac_systems_primarily_serving_comp_rooms_list_x = hvac_systems_primarily_serving_comp_rooms_list_x.append(hvac_x)`          
+        - Check if computer room floor area is greater than 50% of the total floor area served by the hvac system, if yes then carry on with the check: `if hvac_sys_computer_room_floor_area/hvac_sys_total_floor_area > 0.5:` 
+            - Add to the list of hvac systems, hvac_systems_primarily_serving_comp_rooms_list: `hvac_systems_primarily_serving_comp_rooms_list = hvac_systems_primarily_serving_comp_rooms_list.append(hvac)`
```
Comment (Jiarongx-Xie) [inline docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md:30]: I would suggest using `append(hvac_x)` for consistency or using `hvac_id` for all places where it is called for clarity.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms_test.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            "id": "bldg_1",
+            "building_open_schedule": "bldg_open_sched_1",
+            "building_segments": [
+                {
+                    "id": "bldg_seg_1",
+                    "zones": [
+                        {
+                            "id": "zone_1",
+                            "spaces": [
+                                # The overall area ratio of the computer room < 0.5
```
Comment (Jiarongx-Xie) [inline rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms_test.py:20]: should be > 0.5 in the comment

**Review by Jiarongx-Xie — COMMENTED** [review]
Thanks Yun Joon for the dev, I left a few comments.

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.py (hunk 3 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.py:20]: Addressed

File: rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms_test.py (hunk 6 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms_test.py:20]: Good catch. Addressed

File: docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md:22]: Addressed

File: docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md (hunk 5 above)
Comment (yunjoonjung-PNNL) [reply docs/ruleset_functions/get_HVAC_systems_primarily_serving_comp_rooms.md:30]: Good point. I used ```hvac.id``` to be consistent with the earlier part in the RDS.

**Comment by weilixu** [comment]
I thought this is a modification, not developing a new function?

**Comment by yunjoonjung-PNNL** [comment]
> I thought this is a modification, not developing a new function?

Good point! I wasn't aware that this function already exists. I'll close this PR and make updates on the existing function in a new PR.

**Pull request closed without merging**